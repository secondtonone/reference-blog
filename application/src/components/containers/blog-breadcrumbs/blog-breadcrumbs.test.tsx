/**
 * @jest-environment jsdom
 */

import renderer from 'react-test-renderer';
import BlogBreadcrumbs from './blog-breadcrumbs';

jest.mock('~/components/containers/svg-icon', () => ({ code }) => <span>{code}</span>);

describe('Match snapshots BlogBreadcrumbs', () => {
  it('for category', () => {
    const tree = renderer.create(
      <BlogBreadcrumbs />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('for subcategory', () => {
    const tree = renderer.create(
      <BlogBreadcrumbs
        links={[
          {
            url: '/blog/category/seo',
            name: 'SEO'
          }
        ]}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
