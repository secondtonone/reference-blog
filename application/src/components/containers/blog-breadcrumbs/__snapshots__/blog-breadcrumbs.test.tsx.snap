// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshots BlogBreadcrumbs for category 1`] = `
<div
  className="box-adaptive__BoxStyled-sc-g2wy39-0 gfnULt breadcrumbs__BreadcrumbsStyled-sc-1v67b40-0 buCJEi"
  data-test="breadcrumbs"
>
  <ul
    className="text-content__TextStyled-sc-1stkrsz-0 kAbtGH"
    data-test="breadcrumbs-links"
  >
    <li
      data-test="breadcrumbs-link-level-0"
    >
      <a
        href="/blog/"
      >
        Main page /
      </a>
    </li>
  </ul>
</div>
`;

exports[`Match snapshots BlogBreadcrumbs for subcategory 1`] = `
<div
  className="box-adaptive__BoxStyled-sc-g2wy39-0 gfnULt breadcrumbs__BreadcrumbsStyled-sc-1v67b40-0 buCJEi"
  data-test="breadcrumbs"
>
  <ul
    className="text-content__TextStyled-sc-1stkrsz-0 kAbtGH"
    data-test="breadcrumbs-links"
  >
    <li
      data-test="breadcrumbs-link-level-0"
    >
      <a
        href="/blog/"
      >
        Main page /
      </a>
    </li>
    <li
      data-test="breadcrumbs-link-level-1"
    >
      <a
        href="/blog/category/seo/"
      >
        SEO
      </a>
    </li>
  </ul>
</div>
`;
